xnmt/persistence.py

```diff
@@ -535,13 +536,17 @@ def set_child_serializable(node, name, val):
 
 @set_child.register(list)
 def set_child_list(node, name, val):
+  if name == "append": name = len(node)
```

Comment:
What happens if someone accidentally creates a member named "append" in a class they're implementing? Would this cause strange impossible-to-trace errors in the serialization process? From my reading of the code it seems so. If so, I'd suggest naming "append" something that people are less likely to use ("__append"?) and throw an error if you detect that someone has accidentally used this name.

Reply:
The ```@set_child.register(list)``` makes sure that this is only called when ```node``` is a subtype of ```list``` which accepts only integers as keys, so I think this should be safe.